examples/calib_pulses.py

```diff
@@ -0,0 +1,98 @@
+avoid damaging the instrumentation! It is advised that the attenuator be
+calibrated using the GDS and AFG beforehand
+"""
+import pyspecdata as psd
+import os
+import SpinCore_pp as spc
+from datetime import datetime
+from Instruments import GDS_scope
+import numpy as np
+
+calibrating = True
+my_exp_type = "test_equipment"
+assert os.path.exists(psd.getDATADIR(exp_type=my_exp_type))
+# {{{ importing acquisition parameters
+config_dict = spc.configuration("active.ini")
+(
+    nPoints,
+    config_dict["SW_kHz"],
+    config_dict["acq_time_ms"],
+) = spc.get_integer_sampling_intervals(
+    config_dict["SW_kHz"], config_dict["acq_time_ms"]
+)
+# }}}
+# {{{ add file saving parameters to config dict
+config_dict["type"] = "pulse_calib"
+config_dict["date"] = datetime.now().strftime("%y%m%d")
+config_dict["misc_counter"] += 1
+# }}}
+# {{{ ppg
+sqrt_P = config_dict["amplitude"] * np.sqrt(75)  # we have a 75 W amplifier
+desired_beta = np.linspace(0.5 * sqrt_P, 100, 50)
+tx_phases = np.r_[0.0, 90.0, 180.0, 270.0]
+Rx_scans = 1
+datalist = []
+prog_t = []
+# {{{ set up settings for GDS
+with GDS_scope() as gds:
+    gds.reset()
+    gds.CH2.disp = True
+    gds.write(":CHAN1:DISP OFF")
+    gds.write(":CHAN2:DISP ON")
+    gds.write(":CHAN3:DISP OFF")
+    gds.write(":CHAN4:DISP OFF")
+    gds.CH2.voltscal = 100e-3  # set voltscale to 100 mV
+    gds.timescal(50e-6, pos=0)  # set timescale to 50 us
+    gds.write(":CHAN2:IMP 5.0E+1")  # set impedance to 50 ohm
+    gds.write(":TRIG:SOUR CH2")
+    gds.write(":TRIG:MOD NORMAL")  # set trigger mode to normal
+    gds.write(":TRIG:HLEV 7.5E-2")  # used in gds_for_tune which seems reasonable
+    # }}}
+    if calibrating:
+        t_p_range = np.linspace(0.5, 25, 25)
+    else:
+        t_p_range = spc.prog_plen(desired_beta, config_dict["amplitude"])
+    for index, val in enumerate(t_p_range):
+        t_p = val
+        prog_t.append(t_p)
+        spc.configureTX(
+            config_dict["adc_offset"],
+            config_dict["carrierFreq_MHz"],
+            tx_phases,
+            config_dict["amplitude"],
+            nPoints,
+        )
+        acq_time = spc.configureRX(
+            config_dict["SW_kHz"], nPoints, Rx_scans, config_dict["nEchoes"], 1
+        )  # Not phase cycling so setting nPhaseSteps to 1
+        config_dict["acq_time_ms"] = acq_time
+        spc.init_ppg()
+        spc.load(
+            [
+                ("phase_reset", 1),
+                ("delay_TTL", 1.0),
+                ("pulse_TTL", t_p, 0),
+                ("delay", config_dict["deadtime_us"]),
+            ]
+        )
+        spc.stop_ppg()
+        spc.runBoard()
+        datalist.append(gds.waveform(ch=2))
+        spc.stopBoard()
+data = psd.concat(datalist, "t_p").reorder("t")
+data.set_units("t", "s")
+data.set_prop("set_t", t_p_range)
+data.set_prop("desired_betas", desired_beta)
+data.set_prop("acq_params", config_dict.asdict())
```

Comment:
in the processing code, I use `"t_pulse"`  in place of `"t_p"`  it might be easier to just change this.  In either case, you need to label this dimension with the actual pulse lengths that you  programmed in.

Isn't this something I already said?

Reply:
so, actually at the end, your conditional will come into play again.  If you are calibrating, then you  want the axis to be called t_pulse and be labeled with the pulse lengths.  If you are checking, it makes more sense to call it "beta" and label with the desired beta values.